docs/source/how-to-upstream.mdx

```diff
@@ -1,28 +1,61 @@
+Sharing your files and work is a very important aspect of the Hub. The `huggingface_hub` offers several options for uploading your files to the Hub. You can use these functions independently or integrate them into your library, making it more convenient for your users to interact with the Hub. This guide will show you how to:
 
+* Upload a file with [`upload_file`].
+* Upload a folder with [`upload_folder`].
 * Push files with a `commit` context manager.
 * Push files with the [`~Repository.push_to_hub`] function.
 * Upload very large files with [Git LFS](https://git-lfs.github.com/).
-* Push files without Git installed with [`HfApi`]
+* Use [`create_commit`] to push files without Git.
 
 Whenever you want to upload files to the Hub, you need to log in to your Hugging Face account:
 
-1. Log in to your Hugging Face account with the following command:
+- Log in to your Hugging Face account with the following command:
 
    ```bash
    huggingface-cli login
    ```
 
-2. Alternatively, if you prefer working from a Jupyter or Colaboratory notebook, login with [`notebook_login`]:
+- Alternatively, if you prefer working from a Jupyter or Colaboratory notebook, login with [`notebook_login`]:
 
    ```python
    >>> from huggingface_hub import notebook_login
    >>> notebook_login()
    ```
 
-   [`notebook_login`] will launch a widget in your notebook from which you can enter your Hugging Face credentials.
+   [`notebook_login`] launches a widget in your notebook from which you can enter your Hugging Face credentials.
+
+## Upload a file
+
+Once you've created a repository with the [`create_repo`](./how-to-manage#create-a-repository) function, you can upload your files to your repository. Use the [`upload_file`] function to upload a file. 
+
+Specify the path of the file to upload, where you want to upload the file to in the repository, and the name of the repository you want to add the file to. Depending on your repository type, you can optionally set the repository type as a `dataset`, `model` or `space`.
+
+```py
+>>> from huggingface_hub import HfApi
+>>> api = HfApi()
+>>> api.upload_file(path_or_fileobj="/path/to/local/folder/README.md", 
+...                 path_in_repo="README.md", 
+...                 repo_id="username/test-dataset",
+...                 repo_type="dataset",
+... )
+```
+
+## Upload a folder
+
+Use the [`upload_folder`] function to upload a local folder to a repository. Specify the path of the local folder to upload, where you want to upload the folder to in the repository, and the name of the repository you want to add the folder to. Depending on your repository type, you can optionally set the repository type as a `dataset`, `model` or `space`.
```

Comment:
Similar to "upload a file", I would add here that you first need to create a repo before using this function